examples/dynamo/weight_streaming_example.py

```diff
@@ -0,0 +1,159 @@
+    )
+
+# %%
+# Compiler option
+# ----------------------------------
+#
+# enable_weight_streaming=True option and use_strong_types=True are required to build
+# the engine with weight streaming feature. use_strong_types=True option creates a
+# `strongly typed network <https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#strongly-typed-networks>`_ and only float32 precision is allowed in enabled_precisions option
+#
+trt_model = torch_tensorrt.dynamo.compile(
+    llama2_ep,
+    inputs=input_tensors,
+    enabled_precisions={torch.float32},
+    truncate_double=True,
+    device=DEVICE,
+    enable_weight_streaming=True,
+)
+
+# Warm up for 3 iterations
+_ = time_generate(trt_model, input_tensors, osl, 3)
+
+# %%
+# Running with automatic budget size
+# ----------------------------------
+#
+# Once you specify the enable_weight_streaming option, automatic budget size is configured.
+# This automatic size may not always provide the optimal solution because the automatically determined
+# budget lacks insight into the user's specific memory constraints and usage patterns
+weight_streaming_ctx = torch_tensorrt.runtime.weight_streaming(trt_model)
+mean_latency = time_generate(trt_model, input_tensors, osl, 1)
+weight_budget_pct = (
+    weight_streaming_ctx.device_budget / weight_streaming_ctx.total_device_budget * 100
+)
+print(
+    f"Set weight streaming budget as {weight_budget_pct}%. {weight_streaming_ctx.device_budget} bytes out of {weight_streaming_ctx.total_device_budget}. mean latency = {mean_latency} ms"
+)
+
+# %%
+# Running with weight streaming context manager
+# ----------------------------------
+#
+# Weight streaming budget can be limited by using weight streaming context manager.
+# The permissible range for the budget size is from 0 to ctx.total_device_budget.
+# 0 means maximum memory savings occur by using minimum amounts of memory. Value
+# equal to ctx.total_device_budget will disable weight streaming.
+# If multiple trt engines are created, budgets are distributed proportionally
+
+with torch_tensorrt.runtime.weight_streaming(trt_model) as weight_streaming_ctx:
+    # The size of the streamable weights in the engine
+    streamable_budget = weight_streaming_ctx.total_device_budget
+
+    # get automatic weight streaming budget size by using get_automatic_weight_streaming_budget
+    requested_budget = weight_streaming_ctx.get_automatic_weight_streaming_budget()
+    # Set and get the current weight streaming budget for inference
+    weight_streaming_ctx.device_budget = requested_budget
+    mean_latency = time_generate(trt_model, input_tensors, osl, 1)
+    weight_budget_pct = (
+        weight_streaming_ctx.device_budget
+        / weight_streaming_ctx.total_device_budget
+        * 100
+    )
+    print(
+        f"Set auto weight streaming budget as {weight_budget_pct}%. {weight_streaming_ctx.device_budget} bytes out of {weight_streaming_ctx.total_device_budget}. mean latency = {mean_latency} ms"
```

Comment:
Would the values here be same as line 117 ? If we enable_weight_streaming=True but don't set automatic budget explicitly ?